### ISSUE TITLE
Enhance tempwarn

### ISSUE THREAD

**Issue by EvilOlaf** [BanManagement/BanManager#327]
tl;dr: Add an option to add the length of a new temp warn to the period of an earlier warn.

Long version:
The warning is something we use when we don't wanna kick or ban a player immediately so he gets a chance to improve itself and keeps following the rules. 
We don't wanna punish someone for life when he broke the rules and got war permanent warning as it was and is still common procedure. It is a good way to find player the keep breaking the rules though.

This actually can not be traced well with temp warn, because they expire for each itself and to have some serious punishment for receiving multiple warnings their durations need to be very long.

An easier way to separate users which made a mistake once and continual idiocy the warning times could get add up.

For example: Player X received a temporary warning for spam. The period is one week.
He did not break any further rules within this time and became warn-free.

Player Y has some serious mental issues as we all run into from time to time and spams too. Period one week again.
At the next day he spams again and receives another warn. The first warn will be extended to the time when the second warn expires. So he will have two active warnings for the time of the more recent warning.
After about ten days the mental issues strikes again and he spams again and will receive his 3rd warning (either this or the 2nd warn could trigger further punishments) which will extend the period of the older warnings till the point when the 3rd and most recent warning expires.

Please don't blame me too hard for some raggy explanations as I am not a native English speaker :P.

Best regards
Olaf

**Comment by confuser** [comment]
So essentially extend existing temp warnings? I could add an `-e` flag for all temp commands, that would tell BM to extend the existing time rather than override. Would that suffice?

**Comment by EvilOlaf** [comment]
Yes and no. The newer warning that comes after an existing warning might not have the same reason. The first warning could be for spamming, the second one for - whatever - cheating or so.
Not sure if I understood that correctly, but  with `-e`  the reason of the following warnings would be ignore and the only the existing warning would be extended with the existing reason.

**Comment by confuser** [comment]
The reason could be overridden as well, but I think I understand what you're asking for. Whenever a player is warned again whilst they have an active temporary warning, the active warning is then extended to the same time period as the new warning?

**Comment by EvilOlaf** [comment]
Exactly. So it is easier to determine if someone makes trouble over and over and those who made a mistake from time to time.
